doc/lua_api.txt

```diff
@@ -3928,6 +3929,14 @@ Call these functions only at load time!
         * `textlist`: See `minetest.explode_textlist_event`
         * `table`: See `minetest.explode_table_event`
         * `scrollbar`: See `minetest.explode_scrollbar_event`
+		* Special case: `["key_event"]`: Returns a key that is pressed while
+          inside the form. See `keycodes.txt`. This will not return any keys
+          if focus is inside an element that receives key input.
+		* Special case: `["key_mapped"]`: Returns a key that is pressed while
+          inside the form. As opposed to `key_event`, this field returns keys
+          only if they are mapped to an in-game key. The keycode will be the
+          setting name found in `minetest.conf`. This will not return any keys
+          if focus is inside an element that receives key input.
```

Comment:
What's the use-case of knowing the setting name for the key?

Reply:
This (probably) allows performing different actions, for different keys being pressed.

Reply:
- Sneak-clicking something.
- Pressing the drop button to make something drop. (This makes me think, that it would be nice to have the cursor pos, too.
- Controlling something with the real forward, backward, and so on buttons.

All these things wouldn't work well with different keyboard layouts.

Reply:
`key_mapped` values are `keymap_forward`, `keymap_toggle_chat` and so on (see L3146).
You know that some key is mapped, but not which one.

Reply:
As far as I understand the code [here](https://github.com/minetest/minetest/pull/8679/files#diff-074850a88796e13dad12dd7ae1a8d9f7R3143) correctly, it will be eg. `fields = {key_event = "KEY_KEY_D", key_mapped = "keymap_right", ...}`. And ergo, you know the mapped key.

Reply:
*Possible case:*

* Player 1's in-game forward key is `KEY_W`.
* Player 2's in-game forward key is `KEY_UP`.

So, if some code wants to get the in-game forward key, this code
```lua
if fields.key_event == "KEY_W" then
    do_stuff()
end
```
will only work for Player 1.

On the other hand, try this code:
```lua
if fields.key_mapped == "keymap_forward" then
    do_stuff()
end
```
This will work for all players.

Reply:
Ah I mixed it up. The usecases are valid nonetheless. Here's another one: Change key menu. I don't know how the current change keys menu captures keypresses, but this PR would allow for one such menu to be accessible and modifiable in-game.